crates/torii/core/src/processors/event_message.rs

```diff
@@ -0,0 +1,70 @@
+use anyhow::{Error, Result};
+use async_trait::async_trait;
+use dojo_world::contracts::model::ModelReader;
+use dojo_world::contracts::world::WorldContractReader;
+use starknet::core::types::{BlockWithTxs, Event, TransactionReceipt};
+use starknet::providers::Provider;
+use tracing::info;
+
+use super::EventProcessor;
+use crate::processors::MODEL_INDEX;
+use crate::sql::Sql;
+
+#[derive(Default)]
+pub struct EventMessageProcessor;
+
+#[async_trait]
+impl<P> EventProcessor<P> for EventMessageProcessor
+where
+    P: Provider + Send + Sync,
+{
+    fn event_key(&self) -> String {
+        "".to_string()
+    }
+
+    fn validate(&self, event: &Event) -> bool {
+        // we expect at least 3 keys
+        // 1: event selector
+        // 2: model keys, arbitrary length
+        // last key: system key
+        if event.keys.len() < 3 {
+            info!(
+                "invalid keys for event {}: {}",
+                <EventMessageProcessor as EventProcessor<P>>::event_key(self),
+                <EventMessageProcessor as EventProcessor<P>>::event_keys_as_string(self, event),
+            );
```

Comment:
We may remove this print. There's already a trace for all unprocessed events, if no processor matches in the `process_event` function.